core/trino-main/src/main/java/io/trino/sql/rewrite/ShowQueriesRewrite.java

```diff
@@ -414,20 +453,30 @@ protected Node visitShowColumns(ShowColumns showColumns, Void context)
                             aliasedName("data_type", "Type"),
                             aliasedNullToEmpty("extra_info", "Extra"),
                             aliasedNullToEmpty("comment", "Comment")),
-                    from(tableName.getCatalogName(), COLUMNS.getSchemaTableName()),
+                    relation,
                     predicate,
                     ordering(ascending("ordinal_position")));
         }
 
+
         private static <T> Expression getExpression(PropertyMetadata<T> property, Object value)
                 throws TrinoException
         {
             return toExpression(property.encode(property.getJavaType().cast(value)));
         }
 
+        private Cast cast(Expression expression, VarcharType type)
```

Comment:
It seems that these `cast`s are not needed (`SHOW COLUMNS` does not guarantee any particular result type).

If, however, you think they should be there, then I think it would apply to all users of `toExpression`, so the cast can be applied inline on `return new StringLiteral(value.toString());` line there

Reply:
nit: VarcharType -> Type

Reply:
The are needed in case of `null` values. I don't want to change types for other statements.

Reply:
The only other usage is for SHOW SESSION where the type is known.

Reply:
Ok, then I can change type for other statements.

Anyway, I still need to use it in case of `null` values. Possibly I could remove it from column name and column types as these are known to be not null.

Reply:
i'd pass expected Type into `toExpression` and wrap a cast there, for the null case.

anyway, i am thinking that maybe i am asking for an overkill. i don't like the cast but whatever, can stay